### PR TITLE
Added support for unauthenticated proxy and pool attach

### PR DESCRIPTION
**Pull request by tux-o-matic**
Various small improvements
- Specify a proxy name and port only when authentication is not needed.
- Possibility to register and subscribe to a specific pool instead of auto attach.
- Actually making use of the rhsm.conf template which was not used till now.
- Metadata improvement, will improve visibility and increase quality score once deployed on forge.
- Markdown improvement to improve readability of README.

As pointed out in #4, the 'unless' condition of the Exec should be changed since 'grep Subscribed' means 'Not Subscribed' will be a positive check.

### CODE REVIEW

**Comment by tux-o-matic** [comment]
@gerapeldoorn I assume the command line looks for options in the configuration file by default so all proxy_cli things could be removed from init.pp as long as the Exec has a require=>FIle['/etc/rhsm/rhsm.conf'].

**Comment by tux-o-matic** [comment]
@gerapeldoorn Can you tell me what the conflicts are to prevent this PR to be merged?

**Comment by gerapeldoorn** [comment]
@tux-o-matic Main issue is that I'm not in a position anymore to test any changes as I don't have access to a RedHat environment.
@VeriskPuppet also made some extensive edits in a PR, which I merged. This is causing the conflicts.

I'd be happy to put it in the 'Puppet Community repos' as suggested by @VeriskPuppet, but some google-fu didn't get me results. If you can send me a link on it I'd be happy to contribute.

**Comment by tux-o-matic** [comment]
@gerapeldoorn  Maybe you could reach out to them via puppet-community@googlegroups.com

**Comment by bastelfreak** [comment]
hai @tux-o-matic, we moved this project to voxpupuli. I'm currently doing a modulesync in PR #8, I would like to merge this PR after the modulesync. Could you please rebase after the merge of #8?

**Comment by bastelfreak** [comment]
The modulesync PR is now merged, please rebase against the latest master to fix the merge conflicts.

**Comment by tux-o-matic** [comment]
@bastelfreak It's merged with upstream and latest code contributed since my PR was committed.
I had to arrange some things like change 'ensure' on the 'subscription-manager' package resource to a class parameter and not a hard-coded 'latest'. Some IT Ops are not fans of packages updating by themselves without someone triggering it.
Some jobs in the CI failed but I have a feeling that might a general problem, or is it my PR?

**Comment by bastelfreak** [comment]
Please ignore the fails on ruby1.9.3. This is based on a broken gem dependency. Please take a look at the other runs, there are a few wrong indentations. You can ping us in #voxpupuli on freenode if you've any questions.

**Comment by bastelfreak** [comment]
please check the mail address you used in the two last commits, github is unable to recognize it.

File: manifests/init.pp [hunk 1]
```diff
@@ -45,38 +45,52 @@
...
     $proxycli = ''
   }
 
+  if $pool == undef {
+    $command = "/usr/sbin/subscription-manager register --name=\"${::fqdn}\"  --username=\"${rh_user}\" --password=\"${rh_password}\" --auto-attach ${proxycli} && /usr/sbin/subscription-manager repo-override --repo rhel-${::operatingsystemmajrelease}-server-optional-rpms --add=enabled:1 && /usr/sbin/subscription-manager repo-override --repo rhel-${::operatingsystemmajrelease}-server-extras-rpms --add=enabled:1"
+  } else {
+    $command = "/usr/sbin/subscription-manager register --name=\"${::fqdn}\"  --username=\"${rh_user}\" --password=\"${rh_password}\" ${proxycli} && /usr/sbin/subscription-manager attach --pool=${pool}"
+  }
+
   $command = "/usr/sbin/subscription-manager register --force --name=\"${::fqdn}\"  --username=\"${rh_user}\" --password=\"${rh_password}\" --auto-attach ${proxycli}"
```
Comment (bastelfreak) [inline manifests/init.pp]: you've to remove this line. Otherwise puppet tries to reassign it, which won't work.

File: manifests/init.pp [hunk 2]
```diff
@@ -45,38 +45,52 @@
...
+      $proxycli = "--proxy=http://${proxy_hostname}:${proxy_port}"
+    }
   } else {
     $proxycli = ''
   }
 
+  if $pool == undef {
+    $command = "/usr/sbin/subscription-manager register --name=\"${::fqdn}\"  --username=\"${rh_user}\" --password=\"${rh_password}\" --auto-attach ${proxycli} && /usr/sbin/subscription-manager repo-override --repo rhel-${::operatingsystemmajrelease}-server-optional-rpms --add=enabled:1 && /usr/sbin/subscription-manager repo-override --repo rhel-${::operatingsystemmajrelease}-server-extras-rpms --add=enabled:1"
+  } else {
+    $command = "/usr/sbin/subscription-manager register --name=\"${::fqdn}\"  --username=\"${rh_user}\" --password=\"${rh_password}\" ${proxycli} && /usr/sbin/subscription-manager attach --pool=${pool}"
```
Comment (bastelfreak) [inline manifests/init.pp]: the original code contained ```--force```, isn't that needed?

**Comment by tux-o-matic** [comment]
@bastelfreak the Ci jobs are green now for cases using Ruby versions above 1.9.3. I also fixed the author profile of the last commits.

File: manifests/init.pp (hunk 2 above)
Comment (tux-o-matic) [inline manifests/init.pp]: The --force param is unpredictable and has the potential to cause more problems than solving them.

**Pull request merged**